scala/lambdas/ingest-flow-control/src/main/scala/uk/gov/nationalarchives/ingestflowcontrol/Lambda.scala

```diff
@@ -84,32 +82,31 @@ class Lambda extends LambdaRunner[Input, Unit, Config, Dependencies] {
         sourceSystems match
           case Nil => IO.unit // nothing more to do, no one is waiting
           case _ =>
-          val sourceSystemProbabilityMap = buildProbabilityRangesMap(sourceSystems, 1, Map.empty[String, (Int, Int)])
-          val maxRandomValue: Int = sourceSystemProbabilityMap.values.maxBy(_._2)._2
-          val luckyDip = dependencies.randomInt(1, maxRandomValue)
-          val sourceSystemEntry = sourceSystemProbabilityMap.find(eachEntry => eachEntry._2._1 <= luckyDip && eachEntry._2._2 >= luckyDip).get
-          val systemToStartTaskOn = sourceSystemEntry._1
-
-          dependencies.dynamoClient
-            .getItems[IngestQueueTableItem, IngestQueuePartitionKey](List(IngestQueuePartitionKey(systemToStartTaskOn)), config.flowControlQueueTableName)
-            .flatMap { queueTableItem =>
-              val taskToken = queueTableItem.headOption.map(_.taskToken)
-              val queuedAt = queueTableItem.headOption.map(_.queuedAt)
-              if (taskToken.nonEmpty) {
-                dependencies.stepFunctionClient.sendTaskSuccess(taskToken.get) >>
-                  dependencies.dynamoClient
-                    .deleteItems(config.flowControlQueueTableName, List(IngestQueuePrimaryKey(IngestQueuePartitionKey(systemToStartTaskOn), IngestQueueSortKey(queuedAt.get))))
-                    .void
-              } else {
-                val newSystemList = sourceSystems.filter(_.systemName != systemToStartTaskOn)
-                startTaskBasedOnProbability(newSystemList)
+            val sourceSystemProbabilityMap = buildProbabilityRangesMap(sourceSystems, 1, Map.empty[String, Range])
+            val maxRandomValue: Int = sourceSystemProbabilityMap.values.map(_.endExclusive).max
+            val luckyDip = dependencies.randomInt(1, maxRandomValue)
+            val sourceSystemEntry = sourceSystemProbabilityMap.find(sourceSystemProb => sourceSystemProb._2._1 <= luckyDip && sourceSystemProb._2._2 >= luckyDip).get
```

Comment:
I just remembered, by making the tuple a case class,  you can now do this, which makes it even clearer: 

`val sourceSystemEntry = sourceSystemProbabilityMap.find((_, probRange) => probRange.startInclusive <= luckyDip && probRange.endExclusive >= luckyDip).get`